src/sentry/runner/commands/upgrade.py

```diff
@@ -1,12 +1,45 @@
 import click
 from django.conf import settings
+from django.db import connections
 
 from sentry.runner.decorators import configuration
 
 
+def _check_history():
+    connection = connections["default"]
+    cursor = connection.cursor()
+    try:
+        # If the database has 0 tables we are good to go.
+        cursor.execute(
+            """
+            SELECT COUNT(*) AS total FROM information_schema.tables
```

Comment:
Is it ok to assume access to `information_schema.tables`?

Reply:
I could try reading from a table like `auth_user` as a proxy for 'migrations' having been run instead. There could be self-hosted environments where someone has removed access to `information_schema`.